server/src/main/java/io/druid/query/lookup/LookupModule.java

```diff
@@ -185,34 +194,44 @@ public LookupResourceListenerAnnouncer(
 class LookupListeningAnnouncerConfig extends ListeningAnnouncerConfig
 {
   public static final String DEFAULT_TIER = "__default";
+  private final DataSourceNameHolder dataSourceNameHolder;
   @JsonProperty("lookupTier")
   private String lookupTier = null;
```

Comment:
is there any particular  reason that it is not part of the constructor for a design perspective ?